ethabi/src/filter.rs

```diff
@@ -121,21 +119,20 @@ impl<T> From<Topic<T>> for Vec<T> {
 	}
 }
 
-#[cfg(feature = "full-serde")]
+#[cfg(feature = "serde")]
 impl Serialize for Topic<Hash> {
 	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
 	where
 		S: Serializer,
 	{
-		let value = match *self {
-			Topic::Any => Value::Null,
+		match *self {
+			Topic::Any => Option::<()>::None.serialize(serializer),
 			Topic::OneOf(ref vec) => {
-				let v = vec.iter().map(|h| format!("0x{:x}", h)).map(Value::String).collect();
-				Value::Array(v)
+				let v: Vec<_> = vec.iter().map(|h| format!("0x{:x}", h)).collect();
+				v.serialize(serializer)
```

Comment:
AFAIK `Hash: Serialize` - so you should be able to:
```suggestion
				vec.serialize(serializer)
```